sklearn/mixture/gmm.py

```diff
@@ -688,9 +688,10 @@ def _covar_mstep_full(gmm, X, responsibilities, weighted_X_sum, norm,
         post = responsibilities[:, c]
         # Underflow Errors in doing post * X.T are  not important
         np.seterr(under='ignore')
-        avg_cv = np.dot(post * X.T, X) / (post.sum() + 10 * EPS)
-        mu = gmm.means_[c][np.newaxis]
-        cv[c] = (avg_cv - np.dot(mu.T, mu) + min_covar * np.eye(n_features))
+        mu = gmm.means_[c]
+        diff = X - mu
+        avg_cv = np.dot(post * diff.T, diff) / (post.sum() + 10 * EPS)
+        cv[c] = avg_cv + min_covar * np.eye(n_features)
```

Comment:
While not strictly related to this PR, it would be great to use a with block to remove the side effect wrt underflow warnings:

```python
with np.errstate(under='ignore'):
    # Underflow Errors in doing post * X.T are not important
    mu = gmm.means_[c]
    diff = X - mu
    avg_cv = np.dot(post * diff.T, diff) / (post.sum() + 10 * EPS)
    cv[c] = avg_cv + min_covar * np.eye(n_features)